.drone.yml

```diff
@@ -66,3 +66,7 @@ pipeline:
       event: [ push ]
       branch: [ master, releases/*, refs/tags/* ]
       status: success
+
+services:
+  selenium:
+    image: selenium/standalone-firefox
```

Comment:
Instead of running standalone, why not run with selenium grid with nodes? I have done it before and can run whole setup under services (I can get you those docker compose steps). Also, I think we should parallelize these test executions, wouldn't suggest running them serially.

Reply:
Something like this: https://carlosbecker.com/posts/selenium-grid-docker-compose/

Reply:
I do not see how that will work with drone services

Reply:
I think that should work if you replace `selenium` service you have currently with below two services:
```
hub:
  image: selenium/hub
  ports:
    - "4444:4444"
firefox:
  image: selenium/node-firefox
  links:
    - hub
```

Drone will start it up and you will have grid up and running before drone steps.

Reply:
It interleaves services startup so that the hub is not up before the node attempts to start:
```
Running Matrix job #0
[hub:L0:0s] starting selenium hub with configuration:
[hub:L1:0s] {
[hub:L2:0s]   "host": null,
[hub:L3:0s]   "port": 4444,
[hub:L4:0s]   "role": "hub",
[hub:L5:0s]   "maxSession": 5,
[hub:L6:0s]   "newSessionWaitTimeout": -1,
[hub:L7:0s]   "capabilityMatcher": "org.openqa.grid.internal.utils.DefaultCapabilityMatcher",
[hub:L8:0s]   "throwOnCapabilityNotPresent": true,
[hub:L9:0s]   "jettyMaxThreads": -1,
[hub:L10:0s]   "cleanUpCycle": 5000,
[hub:L11:0s]   "browserTimeout": 0,
[hub:L12:0s]   "timeout": 30,
[hub:L13:0s]   "debug": false
[hub:L14:0s] }
[firefox:L0:0s] Not linked with a running Hub container
[hub:L15:0s] 15:49:13.978 INFO - Selenium build info: version: '3.6.0', revision: '6fbf3ec767'
[hub:L16:0s] 15:49:13.980 INFO - Launching Selenium Grid hub
[hub:L17:0s] 2017-10-26 15:49:14.548:INFO::main: Logging initialized @873ms to org.seleniumhq.jetty9.util.log.StdErrLog
[hub:L18:0s] 15:49:14.569 INFO - Will listen on 4444
[hub:L19:0s] 2017-10-26 15:49:14.618:INFO:osjs.Server:main: jetty-9.4.5.v20170502
[hub:L20:0s] 2017-10-26 15:49:14.663:INFO:osjs.session:main: DefaultSessionIdManager workerName=node0
[hub:L21:0s] 2017-10-26 15:49:14.663:INFO:osjs.session:main: No SessionScavenger set, using defaults
[hub:L22:0s] 2017-10-26 15:49:14.666:INFO:osjs.session:main: Scavenging every 660000ms
[hub:L23:0s] 2017-10-26 15:49:14.672:INFO:osjsh.ContextHandler:main: Started o.s.j.s.ServletContextHandler@4c40b76e{/,null,AVAILABLE}
[hub:L24:0s] 2017-10-26 15:49:14.684:INFO:osjs.AbstractConnector:main: Started ServerConnector@123f1134{HTTP/1.1,[http/1.1]}{0.0.0.0:4444}
[hub:L25:0s] 2017-10-26 15:49:14.685:INFO:osjs.Server:main: Started @1010ms
[hub:L26:0s] 15:49:14.685 INFO - Nodes should register to http://172.17.0.3:4444/grid/register/
[hub:L27:0s] 15:49:14.686 INFO - Selenium Grid hub is up and running
[vic-ui:L0:0s] + sleep  15
[vic-ui:L1:14s] + pybot tests/manual-test-cases/Group1-VCH-Creation-Wizard/1-01-Basic-VCH-Create.robot
```

Reply:
regardless, this PR works correctly as it stands.  If we want to upgrade later to a grid model, then we can but that should not stop this PR from going in to allow us to start testing ASAP.

Reply:
sure, we can look at it later.